### PR TITLE
Navigation menu

### PR DESCRIPTION
**Pull request by Scharrre97**
**PT-Story** 
https://www.pivotaltracker.com/story/show/143021805
**Done:**
- Write test for nav menu
- Added navigation menu
- Added links to navigation menu
**Learn:**
- Be better to write test
- How to link i HAML
**Screenshot:**
screen shot 2017-04-05 at 13 54 20

### CODE REVIEW

File: app/views/layouts/application.html.haml [hunk 1]
```diff
@@ -22,10 +20,18 @@
...
+            %li
+              = link_to 'Sign up', new_user_registration_path
             %li
-              %a{:href => "#"} Login
-    
+              = link_to 'Sign in', new_user_session_path
+            %li
+              - if session[:cart_id]
+                = link_to 'Cart', cart_path(session[:cart_id])
+                = @cart.shopping_cart_items.count
```
Comment (tochman) [inline app/views/layouts/application.html.haml]: What IF there's no cart created?

File: app/views/menus/show.html.haml [hunk 2]
```diff
@@ -13,4 +13,4 @@ dishes in cart.
       = submit_tag('Buy')
       = hidden_field_tag(:dish_id, dish.id)
 - if session[:cart_id]
-  = link_to 'Checkout', cart_path(session[:cart_id])
+  = link_to 'Checkout', cart_path(session[:cart_id]), class: 'Checkout'
```
Comment (tochman) [inline app/views/menus/show.html.haml:16]: CSS classes should always start in lowercase

File: app/views/layouts/application.html.haml (hunk 1 above)
Comment (casperjureskog) [reply app/views/layouts/application.html.haml]: did you miss the IF?

Comment (tochman) [reply app/views/layouts/application.html.haml]: Yes I did!

File: app/views/layouts/application.html.haml [hunk 3]
```diff
@@ -22,10 +20,18 @@
...
             %li
-              = link_to 'Registration', '#'
+              = link_to 'Home', root_path
+            %li
+              = link_to 'Sign up', new_user_registration_path
             %li
-              %a{:href => "#"} Login
-    
+              = link_to 'Sign in', new_user_session_path
+            %li
```
Comment (tochman) [inline app/views/layouts/application.html.haml]: This tag should be inside the `if ` statement ;-)

File: features/add_cart.feature [hunk 4]
```diff
@@ -21,10 +21,11 @@ Feature: Add dish to cart
...
     And I should see "Kebabrulle has been added to your order"
-    Then I should see "1 dishes in cart"
+    Then I should see "1 items in cart"
 
   Scenario: I buy a second dish and put it in the cart
     Given I am on the restaurant menu and have one item in my cart
     And I click "Buy" on "Falafel"
     And I should see "Falafel has been added to your order"
-    Then I should see "2 dishes in cart"
+    Then show me the page
```
Comment (tochman) [inline features/add_cart.feature]: Ha!

**Review by tochman — APPROVED** [review]
Good job!

**Pull request merged**